fe/fe-core/src/main/java/com/starrocks/task/TabletTaskExecutor.java

```diff
@@ -70,10 +70,11 @@ public static void buildPartitionsSequentially(long dbId, OlapTable table, List<
         int partitionGroupSize = Math.max(1, numBackends * 200 / Math.max(1, avgReplicasPerPartition));
         boolean enableTabletCreationOptimization = table.isCloudNativeTableOrMaterializedView()
                 && Config.lake_enable_tablet_creation_optimization;
+        long gtid = GlobalStateMgr.getCurrentState().getGtidGenerator().nextGtid();
```

Comment:
I think it is not a good idea to generate a global transaction id here, because it is a low-level function. It is far from the actual meaning of gtid. How can we ensure that a transaction will not call the buildPartition function multiple times? I think it should be placed in the upper-level begin transaction.

Reply:
added option arg to pass the gtid to the low-level function